lib/bundler/rubygems_integration.rb

```diff
@@ -67,6 +67,9 @@ def configuration
       Bundler.ui.error "#{e.class}: #{e.message}"
       Bundler.ui.trace e
       raise Gem::SystemExitException
+    rescue Psych::SyntaxError => e
+      raise YAMLSyntaxError.new(e, "Your RubyGems configuration file (usually located at ~/.gemrc) " \
```

Comment:
Ugh, that's frustrating. Okay, I think saying "The error is in your RubyGems configuration, which is usually located in ~/.gemrc." is the best option we have, then.